crates/red_knot_python_semantic/mdtest.py

```diff
@@ -0,0 +1,209 @@
+"""A runner for Markdown-based tests for Red Knot"""
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#     "rich",
+#     "watchfiles",
+# ]
+# ///
+
+from __future__ import annotations
+
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Final, Literal, Never, assert_never
+
+from rich.console import Console
+from watchfiles import Change, watch
+
+CRATE_NAME: Final = "red_knot_python_semantic"
+CRATE_ROOT: Final = Path(__file__).resolve().parent
+MDTEST_DIR: Final = CRATE_ROOT / "resources" / "mdtest"
+
+
+class MDTestRunner:
+    mdtest_executable: Path | None
+    console: Console
+
+    def __init__(self) -> None:
+        self.mdtest_executable = None
+        self.console = Console()
```

Comment:
minor: you could consider using a dataclass for the `MDTestRunner`. It doesn't help with our `__init__` method much, but it would autogenerate a reasonable `__repr__` method for the class

```suggestion
from dataclasses import dataclass
from pathlib import Path
from typing import Final, Literal, Never, assert_never

from rich.console import Console
from watchfiles import Change, watch

CRATE_NAME: Final = "red_knot_python_semantic"
CRATE_ROOT: Final = Path(__file__).resolve().parent
MDTEST_DIR: Final = CRATE_ROOT / "resources" / "mdtest"


@dataclass
class MDTestRunner:
    mdtest_executable: Path | None = None
    console: Console

    def __init__(self) -> None:
        self.console = Console()
```

Reply:
Hm :smile:. I love dataclasses, but this class feels very much *not* like a data class. It has a custom `__init__`, I don't want/need auto-generated comparison methods, and I probably don't want to debug this class by inspecting it's `repr` (I previously had a logging output that showed the `mdtest_executable` path, but removed that because it didn't add any value). Change my mind.